crates/typst/src/text/raw.rs

```diff
@@ -433,6 +460,15 @@ impl Show for Packed<RawElem> {
         let lines = self.lines().map(|v| v.as_slice()).unwrap_or_default();
 
         let mut seq = EcoVec::with_capacity((2 * lines.len()).saturating_sub(1));
+
+        let dir = RawElem::dir_in(styles);
+        // Insert "Explicit Directional Isolate".
+        match dir {
+            Dir::LTR => seq.push(TextElem::new("\u{2066}".into()).pack()),
+            Dir::RTL => seq.push(TextElem::new("\u{2067}".into()).pack()),
+            _ => {}
```

Comment:
Should we do something with `TTB` and `BTT` (likewise for my previous changes to `inline/mod.rs` #3532)